graphql_compiler/compiler/emit_match.py

```diff
@@ -111,10 +111,10 @@ def _represent_fold(fold_location, fold_ir_blocks):
     mark_name, _ = fold_location.get_location_name()
     base_location_name, _ = fold_location.base_location.get_location_name()
 
-    validate_safe_string(mark_name)
-    validate_safe_string(base_location_name)
-    validate_safe_string(edge_direction)
-    validate_safe_string(edge_name)
+    validate_safe_or_special_string(mark_name)
```

Comment:
Many of these shouldn't be `safe_or_special`, they should just be `safe`. The mark name can never be `@rid`.

I realize that this is something the old code allowed, but now that we are making this in-depth change, leaving that in would just add to the confusion since it seems like an intentional decision to allow these special names.